store/sqlstore/group_store.go

```diff
@@ -1443,15 +1443,23 @@ func (s *SqlGroupStore) GroupMemberCount() (int64, *model.AppError) {
 }
 
 func (s *SqlGroupStore) DistinctGroupMemberCount() (int64, *model.AppError) {
-	return s.countTableWithSelect("COUNT(DISTINCT UserId)", "GroupMembers")
+	return s.countTableWithSelectAndWhere("COUNT(DISTINCT UserId)", "GroupMembers", nil)
+}
+
+func (s *SqlGroupStore) GroupCountWithAllowReference() (int64, *model.AppError) {
+	return s.countTableWithSelectAndWhere("COUNT(*)", "UserGroups", sq.Eq{"AllowReference": true, "DeleteAt": 0})
 }
 
 func (s *SqlGroupStore) countTable(tableName string) (int64, *model.AppError) {
-	return s.countTableWithSelect("COUNT(*)", tableName)
+	return s.countTableWithSelectAndWhere("COUNT(*)", tableName, nil)
 }
 
-func (s *SqlGroupStore) countTableWithSelect(selectStr, tableName string) (int64, *model.AppError) {
-	query := s.getQueryBuilder().Select(selectStr).From(tableName).Where(sq.Eq{"DeleteAt": 0})
+func (s *SqlGroupStore) countTableWithSelectAndWhere(selectStr, tableName string, whereStmt map[string]interface{}) (int64, *model.AppError) {
+	if whereStmt == nil {
+		whereStmt = sq.Eq{"DeleteAt": 0}
+	}
```

Comment:
I'm 0/5 about changing this. It actually provides more versatile functionality and all the callers are right here. 
But shouldn't "DeleteAt" always be added? Based on previous code, it was always added. The caller shouldn't need to add it. Also, if caller passed in an empty (not nil) map. "DeleteAt" wouldn't get added.

Reply:
Yea I was thinking about doing that too but then that removes the possibility of having this function ever return deleted entries (if we ever want that in the future) - I guess I could check if the key exists for `DeleteAt` and only set `DeleteAt = 0` if the key does not exist in the passed where statement